tinygrad/codegen/cstyle.py

```diff
@@ -64,12 +64,12 @@ def render_load(self, output_dtype, buf_name, buf_dtype, idx, local=False) -> st
       return f"vload_half{'' if output_dtype.sz == 1 else str(output_dtype.sz)}(0, {buf_name}+{idx.render(render_cl, strip_parens=True)})"
     if output_dtype.sz > 1:
       return f"({output_dtype.name})(*(({self.smem_prefix if local else self.buffer_prefix}{buf_dtype.name}{output_dtype.sz}*)({buf_name}+{idx.render(render_cl, strip_parens=True)})))"
-    return f"*({buf_name}+{idx.render(render_cl, strip_parens=True)})" if self.uses_ptr_arithmetic else f"{buf_name}[{idx.render(render_cl)}]"
+    return f"*({buf_name}+{idx.render(render_cl, strip_parens=True)})" if self.uses_ptr_arithmetic else f"{buf_name}[{sym_render(idx, render_cl)}]"
```

Comment:
I don't understand why this has to change?

Reply:
yeah my bad, I was passing 0 instead of NumNode(0) in the earlier version. fixed